packages/aws-cdk-lib/aws-codebuild/lib/project.ts

```diff
@@ -1964,6 +1977,18 @@ export enum WindowsImageType {
    * The WINDOWS_SERVER_2019_CONTAINER environment type
    */
   SERVER_2019 = 'WINDOWS_SERVER_2019_CONTAINER',
+
+  /**
+   * The WINDOWS_SERVER_2022_CONTAINER environment type
+   *
+   * @deprecated Windows Server 2022 images are not supported for on-demand compute.
```

Comment:
I don't think we should be adding version that we are immediately marking as deprecated.

Reply:
Yeah, I just want to give the user a warning that they probably won't be able to use the image as is. I thought the `@deprecated` was a good compromise. I can delete them here and we can add them back with #29617's PR

Reply:
@TheRealAmazonKendra Nudge